### DISCUSSION TITLE
Empty Design tab on Mac M1 Pro

### DISCUSSION THREAD

**Discussion by nelo1986** [eclipse/windowbuilder#340] [Q&A] [answered]
Hello, 

I'm trying to use Eclipse and Window Builder to make an SWT project, but and have the following issue.

<img width="800" alt="image" src="https://user-images.githubusercontent.com/95741746/197329659-c7e134f1-cdec-488e-ac65-ab41609ceab6.png">

The file compiles and executes well:

<img width="800" alt="image" src="https://user-images.githubusercontent.com/95741746/197329738-b8fe9d0a-7a73-4343-abd1-3d8f4e419742.png">

I have tried lots of different versions of JDK, Eclipse and WB. Currently, I have the following versions.

**Window Builder**

<img width="300" alt="image" src="https://user-images.githubusercontent.com/95741746/197329857-1f177a8c-29e2-4e01-ab29-aa203a12ee3b.png">

**Eclipse version:** Version: 2022-12 M1 (4.26.0 M1)

**Java:** JDK 18

There aren't errors or warnings, only the issue explained above.

Could you help me?

**Comment by nelo1986** [comment]
I have uninstalled the Eclipse and Dark Theme. Nothing changed. 
In order to give more information, when I create a **Swing -->Application Window** , the Design tab works fine but after Run As --> Java application It doesn't run.

**Design tab on Swing Application Window**
<img width="800" alt="image" src="https://user-images.githubusercontent.com/95741746/197346959-0d7f6dd4-14a6-434a-b572-5ef7922cb71d.png">

**Design tab on SWT Application Window**

<img width="800" alt="image" src="https://user-images.githubusercontent.com/95741746/197347009-b46fb7c7-f422-439c-8d2f-4b7a20afe106.png">

So, I suspect that the problem could be on referenced Libraries which use SWT.

<img width="800" alt="image" src="https://user-images.githubusercontent.com/95741746/197347102-1a461ee2-416d-43a4-83db-88297eeeee01.png">

Information details of WindowBuilder **installed**
<img width="988" alt="image" src="https://user-images.githubusercontent.com/95741746/197347221-93e921c8-15eb-4c25-bd2d-58718c518792.png">

**Comment by howlger** [comment]
This does not look like a bug of WindowBuilder, but a misconfiguration of the _Java Build Path_ of your project for SWT.

Follow the [SWT documentation](https://www.eclipse.org/swt/eclipse.php) (in your case, for macOS on Apple silicon (M1), download and use [`swt-4.25-cocoa-macosx-aarch64.zip`](https://www.eclipse.org/downloads/download.php?file=/eclipse/downloads/drops4/R-4.25-202208311800/swt-4.25-cocoa-macosx-aarch64.zip)).

**Comment by nelo1986** [comment]
Hello @howlger,

Thanks for your help. I followed this [SWT documentation](https://www.eclipse.org/swt/eclipse.php) (I did not find the specific instructions for macOS on Apple silicon (M1)). I have done the following:

I Imported the file [swt-4.25-cocoa-macosx-aarch64.zip](https://www.eclipse.org/downloads/download.php?file=/eclipse/downloads/drops4/R-4.25-202208311800/swt-4.25-cocoa-macosx-aarch64.zip&mirror_id=1190)

<img width="600" alt="iScreen Shoter - 20221023162756611" src="https://user-images.githubusercontent.com/95741746/197398089-4a1ec18b-aabc-4a64-be30-15593db58dcd.png">


<img width="600" alt="iScreen Shoter - 20221023162924533" src="https://user-images.githubusercontent.com/95741746/197398215-db4d5f02-482f-4da9-bd20-2a5c7a425b16.png">

I added it to my current test project (Choosing the Classpath option).

<img width="600" alt="iScreen Shoter - 20221023164010123" src="https://user-images.githubusercontent.com/95741746/197398438-0034744a-2613-4afd-b63d-ec2dc74ee12f.png">

It seems to be running well (but the problem is no running, is with the Design tab).

<img width="600" alt="iScreen Shoter - 20221023163200086" src="https://user-images.githubusercontent.com/95741746/197398239-0d7ed942-59d2-4303-9194-0767c0a1c49b.png">

But the design tab is still empty.
<img width="600" alt="iScreen Shoter - 20221023163221083" src="https://user-images.githubusercontent.com/95741746/197398255-e791059d-a538-4708-b2d0-a5dabc28f813.png">

I don't know what I doing wrong. Thanks again for your help.

**Comment by howlger** [comment]
Your `we.java` file is too simple, as everything related to SWT is in the main method.

[Create a new Java file by choosing something of _File > New > Other...: WindowBuilder > SWT Designer > SWT_](https://www.vogella.com/tutorials/EclipseWindowBuilder/article.html#using-the-swt-designer-windowbuilder-for-visual-ui-design).

**Comment by nelo1986** [comment]
Hello @howlger ,

On **Windows machine**, I created a simple file and everything works:

<img width="600" alt="iScreen Shoter - 20221023174526279" src="https://user-images.githubusercontent.com/95741746/197401829-228f4e02-8b38-43f0-9784-ee9ca6bb408f.png">

The Design Tab is enabled and I can work on my project:
<img width="600" alt="image" src="https://user-images.githubusercontent.com/95741746/197401872-b71d0bd9-b625-4522-b110-6268cc318869.png">

But doing the same on Mac with an Apple silicon chip I can't do anything, because the Design tab is empty.
<img width="600" alt="image" src="https://user-images.githubusercontent.com/95741746/197402060-5ab1caa0-9164-4a70-b4ef-e32b2a9c3257.png">

I created both projects following the next instructions.

1. Create the project
File --> New --> Other --> WindowBuilder -->SWT Designer -->SWT/Jface java Project

2. Create the SWT File
File --> New --> Other --> WindowBuilder -->SWT Designer-->SWT--> Application Window

I followed the same instructions on both machines (Windows and Apple Silicon).

Thanks for your help

**Comment by howlger** [comment]
I see. Could you please provide an Java class with which it can be reproduce on macOS M1, but not on Windows?

Is it reproducible using on macOS with an [x86_64 Eclipse](https://www.eclipse.org/downloads/download.php?file=/technology/epp/downloads/release/2022-09/R/eclipse-java-2022-09-R-macosx-cocoa-x86_64.dmg) running emulated on M1?

Is there something in the error log?

**Comment by nelo1986** [comment]
Hello, 

I have tried with different files. For example [hi.java.zip](https://github.com/eclipse/windowbuilder/files/9850080/hi.java.zip)

But in fact, the file works on both systems. I can run the file, but It can´t be edited using the **Design tab** on MacOs with an Aarch64 processor.

Following your suggestion, I installed an X86_64_Eclipse and It works:

<img width="800" alt="iScreen Shoter - 20221024104426977" src="https://user-images.githubusercontent.com/95741746/197485726-dccddc53-cffc-4aed-94ac-7ee97473c920.png">

It seems that there is a bug with the aarm64 version.

regards

**Comment by howlger** [comment]
The fact that it works on macOS x86_64 but for the same thing not natively on the AArch64 architecture proves that it is indeed a bug. Thank you for providing the proof.

Is there something in the _Error Log_ view?

Could you please report this as a new bug with a minimal Java class as text to reproduce this issue, if possible based on Swing instead of SWT so no dependencies are required.

Maybe you have time to look at the code of WindowBuilder. I guess that there are conditions related to the architecture, which have to be extended for `AArch64`.

**Reply by nelo1986** [reply]
I have created the following issue #344

**Comment by wimjongman** [comment, marked as answer]
This was fixed in 1.12.0 with 

* https://github.com/eclipse/windowbuilder/issues/344
* https://github.com/eclipse/windowbuilder/pull/353

Please install WB 1.12.0 or higher and give it a try.

https://www.eclipse.org/windowbuilder/download.php
